pkg/webhook/vm_validatation.go

```diff
@@ -108,3 +123,67 @@ func (vmValidator *vmDeviceHostValidator) validateDevicesFromSameNodes(vmObj *ku
 
 	return nil
 }
+
+func (vmValidator *vmDeviceHostValidator) validateHostDevices(vmObj *kubevirtv1.VirtualMachine) error {
+	for _, hostDevice := range vmObj.Spec.Template.Spec.Domain.Devices.HostDevices {
+		var (
+			foundInPCI, foundInUSB bool
+			err                    error
+		)
+
+		if foundInPCI, err = vmValidator.validatePCIDevice(hostDevice.DeviceName); err != nil {
+			return err
+		}
+
+		if foundInUSB, err = vmValidator.validateUSBDevice(hostDevice.DeviceName); err != nil {
+			return err
+		}
+
+		if !foundInPCI && !foundInUSB {
+			return fmt.Errorf("hostdevice %s: resource name %s not found in pcidevice and usbdevice cache", hostDevice.Name, hostDevice.DeviceName)
+		}
+	}
+	return nil
+}
+
+func (vmValidator *vmDeviceHostValidator) validatePCIDevice(resourceName string) (found bool, err error) {
```

Comment:
these validators check if there is a pci resource that exists in the cluster. Do we need to check the resource with a node in any situation?

Reply:
I think we need to check below situations.

1. The node selector is different from pci/usb device's node
2. The resource is used by other VMs

Currently, we don't have those checks in backend code.

Reply:
I think we could keep this issue simple. So, I'll only check if the resource name exists in the cluster. For your case, if we decide to check node selector, we should consider node affinity. Besides, it's valid that same resource names are used by different VMs. We also need to consider checking if the resource is enough.